File: src/main/java/bfs/PrintBinaryTree.java
package bfs;


import structure.BinaryTree;
import structure.BinaryTree.TreeNode;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

/**
 * https://leetcode-cn.com/problems/cong-shang-dao-xia-da-yin-er-cha-shu-ii-lcof/
 */
public class PrintBinaryTree {
    public static void main(String[] args) {
        Integer[] arr = {5, 3, 9, 7, 1, 6};
        BinaryTree binaryTree = new BinaryTree(arr);
        List<List<Integer>> lists = new PrintBinaryTree().levelOrder(binaryTree.root);
        for (List<Integer> list : lists) {
            System.out.println(list);
        }
    }

    public List<List<Integer>> levelOrder(TreeNode root) {
        LinkedList<List<Integer>> res = new LinkedList<>();
        if (root == null){
            return res;
        }

        LinkedList<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()){
            ArrayList<Integer> list = new ArrayList<>();
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode removeNode = queue.poll();
                list.add(removeNode.val);
                if (removeNode.left != null) queue.add(removeNode.left);
                if (removeNode.right != null) queue.add(removeNode.right);

            }
            res.add(list);
        }
        return res;
    }
}

File: PyAlgorithms/bfs/print_binary_tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from dfs.tree import TreeNode
from dfs.tree import BinaryTree


class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        res = []
        if root is None:
            return res
        queue = []
        queue.append(root)
        while len(queue) != 0:
            sub_res = []
            length = len(queue)
            for i in range(length):
                node = queue.pop(0)
                sub_res.append(node.val)
                if node.left is not None:
                    queue.append(node.left)
                if node.right is not None:
                    queue.append(node.right)
            res.append(sub_res)
        return res


if __name__ == '__main__':
    l = [3, 9, 20, None, None, 15, 7]
    t = BinaryTree(l)
    s = Solution()
    print(s.levelOrder(t.root))
